contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -143,6 +148,24 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 			applyPreimage(preimage)
 			return &h.htlcSuccessResolver, nil
 
+		case <-hodlSubscription.CancelChan:
+			// No action required, but we need to receive to prevent
+			// the channel from getting jammed.
+
+		case preimage := <-hodlSubscription.SettleChan:
```

Comment:
To prevent mixing up preimages of invoices and forwards. Otherwise I could just as well have used the preimage cache in the link, my initial approach.